test/os.cc

```diff
@@ -0,0 +1,87 @@
+#include "eventuals/os.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace eventuals::test {
+namespace {
+
+#if defined(__linux__) || defined(__MACH__)
+
+using eventuals::os::Thread;
+using testing::MockFunction;
+
+void foo() {
+  // Do nothing.
+}
+
+static bool functor_was_called = false;
+
+struct Moveable {
+  Moveable() = default;
+  ~Moveable() = default;
+  Moveable(const Moveable& c) = delete;
+  Moveable& operator=(const Moveable&) = delete;
+  Moveable(Moveable&&) {}
+
+  void operator()() {
+    functor_was_called = true;
+  }
+};
+
+TEST(Thread, NotJoinable) {
+  Thread t;
+  EXPECT_FALSE(t.is_joinable());
+}
+
+TEST(Thread, Joinable) {
+  Thread t{&foo, "thread_name"};
+  EXPECT_TRUE(t.is_joinable());
+  t.join();
+  EXPECT_FALSE(t.is_joinable());
+}
+
+TEST(Thread, LambdaThatCapturesEverything) {
```

Comment:
This test is doing two things, it's showing you can capture things in the lamdba and it's also showing that you can set the stack size. Separate this into two independent threads. And in the test that is setting the stack size, call `GetStackSize()` in the body to assert that the stack size has been successfully set. If the PR with `GetStackSize()` is not completed then let's complete that first.